sympy/algebras/quaternion.py

```diff
@@ -51,10 +51,7 @@ class Quaternion(Expr):
     is_commutative = False
 
     def __new__(cls, a=0, b=0, c=0, d=0, real_field=True):
-        a = sympify(a)
-        b = sympify(b)
-        c = sympify(c)
-        d = sympify(d)
+        a, b, c, d = sympify(a), sympify(b), sympify(c), sympify(d)
```

Comment:
```suggestion
        a, b, c, d = map(sympify, (a, b, c, d))
```

Reply:
Yes, one of multiple ways of doing it... 
```
a, b, c, d = [sympify(e) for e in (a, b, c, d)]
```